mdaviz/mda_file_table_model.py

```diff
@@ -0,0 +1,219 @@
+"""
+QAbstractTableModel of folder content.
+
+.. autosummary::
+
+    ~MDAFileTableModel
+"""
+from mda import readMDA
+from PyQt5 import QtCore
+import yaml
+from . import utils
+
+
+# For this purpose, p and d will all be called detectors
+# since you can plot d vs p, but also d vd d
+
+
+class MDAFileTableModel(QtCore.QAbstractTableModel):
+    checkboxToggled = QtCore.pyqtSignal(int, int)
```

Comment:
Might not be necessary since `setData()` is called automatically by the view when a checkbox is toggled.  See comment below about redesign.

Reply:
Why not a great fit?  As the user selects checkboxes, the selection rules mean that other boxes could change.  If this code emits a `dataChanged` signal for each checkbox change from a selection rule, that emit could cause thrashing between the intermediate checkbox states until evaluation of selection rules is complete.  Looks like extra logic in the code to handle the internal data structure of `self._checkbox_states`.